src/js/components/FrameworkConfiguration.js

```diff
@@ -0,0 +1,381 @@
+import React, { PropTypes, Component } from "react";
+import { Confirm } from "reactjs-components";
+import FullScreenModal from "#SRC/js/components/modals/FullScreenModal";
+import FullScreenModalHeader
+  from "#SRC/js/components/modals/FullScreenModalHeader";
+import FullScreenModalHeaderActions
+  from "#SRC/js/components/modals/FullScreenModalHeaderActions";
+import FullScreenModalHeaderTitle
+  from "#SRC/js/components/modals/FullScreenModalHeaderTitle";
+import FullScreenModalHeaderSubTitle
+  from "#SRC/js/components/modals/FullScreenModalHeaderSubTitle";
+import ToggleButton from "#SRC/js/components/ToggleButton";
+import HashMapDisplay from "#SRC/js/components/HashMapDisplay";
+import ModalHeading from "#SRC/js/components/modals/ModalHeading";
+import UniversePackage from "#SRC/js/structs/UniversePackage";
+import Icon from "#SRC/js/components/Icon";
+import Util from "#SRC/js/utils/Util";
+import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
+import FrameworkConfigurationForm
+  from "#SRC/js/components/FrameworkConfigurationForm";
+import deepEqual from "deep-equal";
+
+export default class FrameworkConfiguration extends Component {
+  constructor(props) {
+    super(props);
+
+    const schema = props.packageDetails.config;
+    const activeTab = Object.keys(schema.properties)[0];
+    const firstField = Object.keys(schema.properties[activeTab].properties)[0];
+    const focusFieldPath = [activeTab, firstField];
+
+    this.state = {
```

Comment:
Not blocking as this is very much  in line with the `CreateServiceModal` component but we need to refactor the way we're handling state as the great number of entangled properties is highly error prone. For details please see [DCOS-13173](https://jira.mesosphere.com/browse/DCOS-13173).

Reply:
@orlandohohmeier I agree and am not satisfied with complexity of state/props in these components. I read that issue, but can you elaborate / give-example of what this kind of "state machine" would look like?